### PR TITLE
fix gdal translate when translating multi band rasters and changing the datatype

### LINKED ISSUE #31544: Translate (Convert Format) produces a blank image
When I trying to convert a Landsat-8 imagery from UInt16 (Unsigned Integer 16 Bit) to Byte (Unsigned Integer 8 Bit), the algorithm produces a blank image.

This video shows the issue: 
https://drive.google.com/file/d/1MnEaYDVF4b4363RTN05ZLsNb4UyHj-aT/view?usp=sharing

**Another question:** I was unable to include the parameter **-scale** to stretch data correcty as I do via GDAL command prompt

QGIS version 3.4.11 64 Bit, Operational System Windows 10 Pro.

### PR DESCRIPTION
**Pull request by gioman**
fixes #31544

### CODE REVIEW

File: python/plugins/processing/algs/gdal/translate.py [hunk 1]
```diff
@@ -72,6 +73,9 @@ def initAlgorithm(self, config=None):
         self.addParameter(QgsProcessingParameterBoolean(self.COPY_SUBDATASETS,
                                                         self.tr('Copy all subdatasets of this file to individual output files'),
                                                         defaultValue=False))
+        self.addParameter(QgsProcessingParameterBoolean(self.SCALE,
```
Comment (nyalldawson) [inline python/plugins/processing/algs/gdal/translate.py]: Either "input pixel values" or "input pixel's value"

**Review by nyalldawson — COMMENTED** [review]
nice! Can you add a unit test covering this to https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/GdalAlgorithmsRasterTest.py#L132 ?

File: python/plugins/processing/algs/gdal/translate.py (hunk 1 above)
Comment (gioman) [reply python/plugins/processing/algs/gdal/translate.py]: Right (I just copied the wording directly from the GDAL docs).

**Comment by alexbruy** [comment]
Do we really need this new parameter? All GDAL algorithms now have an optional "Additional command-line parameters" parameter, which was added for cases when one need to pass a command-line argument(s) not exposed in the algorithm UI.

**Comment by gioman** [comment]
> Do we really need this new parameter? All GDAL algorithms now have an optional "Additional command-line parameters" parameter, which was added for cases when one need to pass a command-line argument(s) not exposed in the algorithm UI.

@alexbruy Hi Alex, isn't that only for "layer creation options" (-lco)?

**Comment by alexbruy** [comment]
No. I have added parameter to all GDAL tools exactly for passing options not exposed in the UI (like this one and in #31692). There is "Additional creation options" parameter for -lco and "Additional command-line parameters" for passing command line arguments.

**Comment by gioman** [comment]
> No. I have added parameter to all GDAL tools exactly for passing options not exposed in the UI (like this one and in #31692). There is "Additional creation options" parameter for -lco and "Additional command-line parameters" for passing command line arguments.

not stale. On second thoughts I think that exposing this parameter would be a valuable addition for the users.

**Comment by gioman** [comment]
Not stale. This should be merged.

**Comment by nyalldawson** [comment]
@gioman it's still missing tests

**Pull request closed without merging**